src/libsyntax_pos/lib.rs

```diff
@@ -159,6 +159,18 @@ impl Span {
         Span::new(BytePos(lo), BytePos(lo), span.ctxt)
     }
 
+    /// Returns a new span representing the previous character after the start-point of this span
+    pub fn prev_point(self) -> Span {
+        let span = self.data();
+        let span_lo = span.lo.0;
+        let lo = if span_lo == 0 {
+            0
+        } else {
+            span_lo - 1
```

Comment:
Doesn't this mean we now have the same problem with the following case:
```rust
let x = &&&0;
let &&x = x;
```

IIUC, the error message would suggest using `&&`, right?
```